tantar/src/main/java/org/opencadc/tantar/BucketValidator.java

```diff
@@ -666,4 +673,44 @@ public Artifact next() {
             }
         };
     }
+
+    /**
+     * Check the StorageMetadata for a matching ObsoleteStorageLocation and if found
+     * delete the file from storage and delete the ObsoleteStorageLocation.
+     */
+    void checkObsoleteStorageLocation(StorageMetadata storageMetadata) throws Exception {
+        ObsoleteStorageLocation obsoleteStorageLocation =
+            this.obsoleteStorageLocationDAO.get(storageMetadata.getStorageLocation());
+        if (obsoleteStorageLocation != null) {
+
+            delete(storageMetadata);
+
+            TransactionManager transactionManager = this.artifactDAO.getTransactionManager();
+            try {
+                LOGGER.debug("start transaction...");
+                transactionManager.startTransaction();
+                LOGGER.debug("start transaction... OK");
+
+                this.obsoleteStorageLocationDAO.delete(obsoleteStorageLocation.getID());
+                LOGGER.debug(String.format("deleted %s", obsoleteStorageLocation));
+
+                LOGGER.debug("commit transaction...");
+                transactionManager.commitTransaction();
+                LOGGER.debug("commit transaction... OK");
```

Comment:
This looks like it will still get executed in `reportOnly` mode.  Does this method need a `canTakeAction()` check?

Reply:
Yes it does need to check canTestAction() before executing. Updated.